src/app/molecules/space-options/SpaceOptions.jsx

```diff
@@ -28,11 +30,19 @@ import { confirmDialog } from '../confirm-dialog/ConfirmDialog';
 
 function SpaceOptions({ roomId, afterOptionSelect }) {
   const mx = initMatrix.matrixClient;
+  const { roomList } = initMatrix;
   const room = mx.getRoom(roomId);
   const canInvite = room?.canInvite(mx.getUserId());
   const isPinned = initMatrix.accountData.spaceShortcut.has(roomId);
   const isCategorized = initMatrix.accountData.categorizedSpaces.has(roomId);
 
+  const handleMarkAsRead = () => {
+    const spaceChildren = roomList.getSpaceChildren(roomId);
```

Comment:
Currently `getSpaceChildren` give one level deep children aka subspaces and rooms, marking them as read does remove notification from child rooms but doesn't do same for child spaces.

We can use `roomList.getCategorizedSpaces([spaceId])` which maps passed spaces and all its child spaces to their children.

Reply:
@ajbura Force pushed a fix in c4a821ceedc7c19d5c483804aa5fc210fe8b176a.